editions/tw5.com/tiddlers/howtos/Concatenating text and variables using macro substitution.tid

```diff
@@ -0,0 +1,57 @@
+created: 20160424150551727
+modified: 20161223225338631
+tags: Learning
+title: Concatenating text and variables using macro substitution
+type: text/vnd.tiddlywiki
+
+It's a frequent use case in ~TiddlyWiki that you will want to put the results of variables together with various bits of strings of text. This process in some programming languages is often referred to as "concatenating" text.
+
+You might, for instance want to set up a template for your customer database, where links will automatically refer to additional contact information about your customer. Inside your tiddler, you might try something like this:
+
+```
+* [[Additional Info|<<currentTiddler>>-Contact]]
+```
+
+But that won't work. If you try this, the link will be interpreted very literally, and will attempt to take you to:
+
+```
+<<currentTiddler>>-Contact
+```
+
+The solution is to use a macro to put the rendered value of `<<currentTiddler>>` together with the bit of additional text, `-Contact`.
+
+Create a macro either at the top of the tiddler or in a global tiddler (see [[Macros]]) like this:
+
+```
+\define linkup(link) [[Additional Info|$link$-Contact]]
```

Comment:
Please always start with the correct syntax and later explain common pitfalls.

```
\define linkup() [[Additional Info|$(currentTiddler)$-Contact]]
```

You should **never** pass **currentTiddler** as a parameter, all that achieves is unwanted complexity.